website/src/Pages/ZinesPage/ZineDetailsPage.jsx

```diff
@@ -0,0 +1,247 @@
+
+  const getZineDetails = async () => {
+    const zine = await axios.get(`http://localhost:4000/zine/getzine/${id}`);
+    setDetails({
+      url: zine.data.url,
+      title: zine.data.title,
+      season: zine.data.season,
+      year: zine.data.year,
+      contents: zine.data.contents,
+    });
+    setTitle(zine.data.title);
+    setSeason(zine.data.season);
+    setYear(zine.data.year);
+    setContents(zine.data.contents);
+  };
+
+  useEffect(() => {
+    getZineDetails();
+  }, [update]);
+
+  const handleDelete = async () => {
+    await axios.delete(`http://localhost:4000/zine/delete/${id}`);
+    navigate('/zines');
+  };
+
+  const handleDeleteSection = (index) => {
+    const newContents = [...contents.slice(0, index), ...contents.slice(index + 1)];
+    setContents(newContents);
+  };
+
+  const handleEditSection = (index, value, part) => {
+    const updatedContents = [...contents];
+    if (part === 'Chapter') {
+      updatedContents[index].sectionTitle = value;
+    } else {
+      updatedContents[index].sectionPage = value;
+    }
+    setContents(updatedContents);
+  };
+
+  const handleAddSection = () => {
+    setContents([...contents, { sectionTitle: '', sectionPage: '' }]);
+  };
+
+  const handleCancel = async () => {
+    setTitle(details.title);
+    setSeason(details.season);
+    setYear(details.year);
+    setContents(details.contents);
+    setEdit(false);
+  };
+
+  const handleSave = async () => {
+    // TODO: handle edge case where someone adds a bunch of empty slots
+    await axios.patch(`http://localhost:4000/zine/update/${id}`, {
+      title,
+      season,
+      year,
+      contents,
+    });
+    setUpdate((val) => val + 1);
+    setEdit(false);
+  };
+
+  return (
+    <Box sx={{ backgroundColor: '#C4C8CA' }}>
+      <Box sx={{ backgroundColor: '#FFFFFF', margin: '3%' }}>
+        <Box sx={{ display: 'flex' }}>
+          <Box>
+            <Typography variant="h5">
+              Zine Title
+            </Typography>
+            {edit ? (
+              <TextField
+                sx={{ width: 400 }}
+                value={title}
+                onChange={(e) => setTitle(e.target.value)}
+              />
+            ) : (
+              <Typography variant="h6">
+                {details.title}
+              </Typography>
+            )}
+          </Box>
+          <Box sx={{ marginLeft: 'auto' }}>
+            {edit ? (
+              <>
+              </>
+            ) : (
+              <Box>
+                <Button variant="contained" onClick={() => setEdit(true)}>
+                  Edit
+                </Button>
+                <Button variant="contained" onClick={() => handleDelete()}>
+                  Delete
+                </Button>
+              </Box>
+            )}
+          </Box>
+        </Box>
+        <Box>
+          <Typography>
+            Edition
+          </Typography>
+          {edit ? (
+            <Box>
+              <TextField
```

Comment:
If we know that the Zines are always denotes by Season and year, then it might be worthwhile to make a dropdown menu for both options, rather than text fields

Reply:
Maybe for the year one, we use datetime (or wtvr JS uses) and just generate a list of options from some arbitrary year (probably when Zines started) until the current year + 1